doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+    }
+```
+Example for interface Ethernet:
+```
+    "INTF_TABLE": {
+        "Ethernet248": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:11",
+            "mpls": "enable"
+        }
+    }
+```
+Example for interface port-channel:
+```
+    "INTF_TABLE": {
+        "PortChannel1": {
+            "loopback_action": "drop",
+            "mac_addr": "00:01:02:03:04:12",
+            "mpls": "enable"            
+        }
+    }
+```
+
+### 2.2.3 Configuration migration
+No special handling is required
+
+## 2.3 Flows
+
+### 2.3.1 Loopback action set interface vlan
+#### 2.3.1.1 Handle set in runtime
+![interface vlan](images/interface_vlan.svg "Figure 2: interface vlan")
+###### Figure 2: interface vlan
+
+#### 2.3.1.2 Handle set in init
+In init, loopback action configuration is being handled as part of the IP interface creation.
+During init, all field-value pairs of each key in INTF_TABLE table (e.g. INTF_TABLE|Vlan100) are being received in intfOrch together at once.
+Those field-value pairs are the interface attributes, e.g. mac address, mpls enable, loopback action etc.
+
+### 2.3.2 Loopback action set interface Ethernet
+Loopback action set on interface Ethernet flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.3 Loopback action set interface port-channel
+Loopback action set on interface port-channel flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.4 IP interface creation
+If loopback action was not configured by the user, default loopback action is derived from SAI behavioral model which is forward.
+
+## 2.4 SAI
+The following will be used for setting IP interface loopback action.
+
+SAI API:
+```
+sai_router_intfs_api->set_router_interface_attribute()
+sai_router_intfs_api->create_router_interface()
+```
+
+SAI attribute:
+###### Table 2: SAI attribute
+| Loopback action  | SAI attribute                                     | Attribute value            |
+|------------------|---------------------------------------------------|----------------------------|
+| drop             | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_DROP     |
+| forward          | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_FORWARD  |
+
+
+```
+typedef enum _sai_router_interface_attr_t
+{
+    /**
+     * @brief Packet action when a packet ingress and gets routed on the same RIF
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_FORWARD
+     */
+    SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION,
+
+} sai_router_interface_attr_t;
+```
+
+## 2.5 Statistics
+Packets that will dropped due to loopback action will be counted in RX_ERR in IP interface statistics.
+```
+root@sonic:~# show interfaces counters rif
```

Comment:
is this correct also for vlan interface and port channel? just making sure.
if so, i suggest your example will include other types of inerfaces

Reply:
Done.